plugin/plugin.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2024 the Drone Authors. All rights reserved.
+// Use of this source code is governed by the Blue Oak Model License
+// that can be found in the LICENSE file.
+
+package plugin
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type ZipPlugin struct {
+	Source   string `envconfig:"PLUGIN_SOURCE"`
+	Target   string `envconfig:"PLUGIN_TARGET"`
+	LogLevel string `envconfig:"PLUGIN_LOG_LEVEL"`
+}
```

Comment:
let's also add `PLUGIN_FORMAT` as a setting and default it to zip as mentioned in the specs

Reply:
I will add PLUGIN_FORMAT as soon as I add multiple archive features - tar and gzip